addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -0,0 +1,413 @@
+	if err != nil {
+		clusterbootstraplog.Error(err, "Error creating aggregated API Resources client")
+		return err
+	}
+
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&runv1alpha3.ClusterBootstrap{}).
+		WithDefaulter(wh).
+		WithValidator(wh).
+		Complete()
+}
+
+//+kubebuilder:webhook:verbs=create;update,path=/validate-run-tanzu-vmware-com-v1alpha3-clusterbootstrap,mutating=false,failurePolicy=fail,groups=run.tanzu.vmware.com,resources=clusterbootstraps,versions=v1alpha3,name=vclusterbootstrap.kb.io
+//+kubebuilder:webhook:path=/mutate-run-tanzu-vmware-com-v1alpha3-clusterbootstrap,mutating=true,failurePolicy=fail,groups=run.tanzu.vmware.com,resources=clusterbootstraps,verbs=create;update,versions=v1alpha3,name=mclusterbootstrap.kb.io
+
+var _ webhook.CustomDefaulter = &ClusterBootstrap{}
+var _ webhook.CustomValidator = &ClusterBootstrap{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (wh *ClusterBootstrap) Default(ctx context.Context, obj runtime.Object) error {
+	// TODO: Placeholder for future work https://github.com/vmware-tanzu/tanzu-framework/issues/1916
+	return nil
+}
+
+func getFieldPath(fieldName string) *field.Path {
+	return field.NewPath("spec").Child(fieldName)
+}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (wh *ClusterBootstrap) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+
+	clusterBootstrap, ok := obj.(*runv1alpha3.ClusterBootstrap)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a ClusterBootstrap but got a %T", obj))
+	}
+	clusterbootstraplog.Info("validate create", "name", clusterBootstrap.Name)
+
+	var allErrs field.ErrorList
+	// Iterating one by one because we need field info for getFieldPath, and some core packages can be nil
+
+	if err := wh.validateClusterBootstrapPackage(ctx, clusterBootstrap.Spec.CNI, clusterBootstrap.Namespace, getFieldPath("cni")); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	if err := wh.validateClusterBootstrapPackage(ctx, clusterBootstrap.Spec.Kapp, clusterBootstrap.Namespace, getFieldPath("kapp")); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	// CSI and CPI can be nil
+	if clusterBootstrap.Spec.CSI != nil {
+		if err := wh.validateClusterBootstrapPackage(ctx, clusterBootstrap.Spec.CSI, clusterBootstrap.Namespace, getFieldPath("csi")); err != nil {
+			allErrs = append(allErrs, err)
+		}
+	}
+	if clusterBootstrap.Spec.CPI != nil {
+		if err := wh.validateClusterBootstrapPackage(ctx, clusterBootstrap.Spec.CPI, clusterBootstrap.Namespace, getFieldPath("cpi")); err != nil {
+			allErrs = append(allErrs, err)
+		}
+	}
+
+	if clusterBootstrap.Spec.AdditionalPackages != nil {
+		// validate additional packages
+		for _, pkg := range clusterBootstrap.Spec.AdditionalPackages {
+			if err := wh.validateClusterBootstrapPackage(ctx, pkg, clusterBootstrap.Namespace, getFieldPath("additionalPackages")); err != nil {
```

Comment:
Yes, I also noticed it when I was implementing it. Since it's an array, looks like the only thing we can do is to include the index.

Added the index in the fieldPath